source/common/http/filter_manager.h

```diff
@@ -55,6 +59,28 @@ class LocalReplyOwnerObject : public StreamInfo::FilterState::Object {
   const std::string filter_config_name_;
 };
 
+// TODO(wbpcode): Rather than allocating every filter with an unique pointer, we may could
+// construct the filter in place in the vector. This should reduce the heap allocation and
+// memory fragmentation.
+struct StreamDecoderFilters {
+  using Element = ActiveStreamDecoderFilter;
+  using Iterator = std::vector<ActiveStreamDecoderFilterPtr>::iterator;
+
+  Iterator begin() { return entries.begin(); }
+  Iterator end() { return entries.end(); }
+
+  std::vector<ActiveStreamDecoderFilterPtr> entries;
+};
+struct StreamEncoderFilters {
```

Comment:
nit: please add comments to these classes, and specifically mention why the direction of the encoder is reversed.

Reply:
done.